test/users/npadmana/fftw/testFFTW.chpl

```diff
@@ -131,9 +134,20 @@ proc runtest(param ndim : int, fn : string) {
 
 }
 
+// Initialize multi-threaded support if desired
+if (usethread) {
+  if (init_threads() == 0) then halt("Failed to initialize thread support:");
+  plan_with_nthreads(nthread);
+}
+
+
 writeln("1D");
 runtest(1, "arr1d.dat");
 writeln("2D");
 runtest(2, "arr2d.dat");
 writeln("3D");
 runtest(3, "arr3d.dat");
+
+if (usethread) {
+  cleanup_threads();
+}
```

Comment:
I didn't realize how clean adding multithreaded support would be!  It looks nice (and I'm planning on merging), but had some questions that I'll put here since it's near the code:

1) Do you think there's any chance that, being a parallel language, Chapel users will just always want multithreaded support enabled?  (In which case, the init_threads() call could be put into FFTW.chpl itself...  We could even put the config param saying whether or not to multithread it there and have it deault to on but permit people to compile with it off if they preferred...  My thinking is "Why put this on the user?"

2) Related, seems like having nthread default to something like here.maxTaskPar might be better than 1.  :)

3) What are the impacts of not cleaning up?  I.e., is it something one's supposed to do on program exit (like closing files) but that failure to do so will just cause the OS to do it?  Or is it worse than that?  We've talked about having a module de-initialize hook, but don't yet (which means if we pushed the init into the module, it'd be hard for us to clean up.  But at present we also won't be calling cleanup on any other exit from a Chapel program like a halt or exit...